ccl/multiregionccl: TestMrSystemDatabase failed

ccl/multiregionccl.TestMrSystemDatabase [failed](https://tanzanite.cluster.engflow.com/invocations/default/46638466-7321-4269-a02c-7b2e1bb7a63c?testReportRun=2&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on release-24.2.0-rc @ [953d09cce5a01cc0053e138516d80a27454f027c](https://github.com/cockroachdb/cockroach/commits/953d09cce5a01cc0053e138516d80a27454f027c):


```
=== RUN   TestMrSystemDatabase
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestMrSystemDatabase1200750091
    test_log_scope.go:81: use -show-logs to present logs inline
    multiregion_system_table_test.go:382: -- test log scope end --
test logs left over in: outputs.zip/logTestMrSystemDatabase1200750091
--- FAIL: TestMrSystemDatabase (85.97s)
=== RUN   TestMrSystemDatabase/Sqlliveness_system_database
    multiregion_system_table_test.go:115: error scanning '&{0xc00afa7680 <nil>}': pq: unsupported comparison: bytes to crdb_internal_region
    --- FAIL: TestMrSystemDatabase/Sqlliveness_system_database (0.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMrSystemDatabase.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41204

--- a/pkg/ccl/multiregionccl/multiregion_system_table_test.go
+++ b/pkg/ccl/multiregionccl/multiregion_system_table_test.go
@@ -108,12 +108,28 @@ func TestMrSystemDatabase(t *testing.T) {
 	}
 	for _, testCase := range testCases {
 		t.Run(fmt.Sprintf("Sqlliveness %s", testCase.name), func(t *testing.T) {
-			row := testCase.database.QueryRow(t, `SELECT crdb_region, session_id, expiration FROM system.sqlliveness LIMIT 1`)
-			var sessionID string
-			var crdbRegion string
-			var rawExpiration apd.Decimal
-			row.Scan(&crdbRegion, &sessionID, &rawExpiration)
-			require.Equal(t, "us-east1", crdbRegion)
+			// When optimizing the system database the ALTER DATABASE command will
+			// delete stats, but these are refreshed in memory using a range feed.
+			// Since there can be a delay in the new stats being picked up its possible
+			// for this query to fail with:
+			// "unsupported comparison: bytes to crdb_internal_region"
+			// querying table statistics. This is a transient condition that will
+			// clear up once the range feed catches up.
+			testutils.SucceedsSoon(t, func() error {
+				row := testCase.database.DB.QueryRowContext(ctx, `SELECT crdb_region, session_id, expiration FROM system.sqlliveness LIMIT 1`)
+				var sessionID string
+				var crdbRegion string
+				var rawExpiration apd.Decimal
+				err := row.Scan(&crdbRegion, &sessionID, &rawExpiration)
+				if err != nil {
+					return err
+				}
+				if crdbRegion != "us-east1" {
+					return errors.AssertionFailedf("unexpected region, got: %q expected: %q",
+						crdbRegion, "us-east1")
+				}
+				return nil
+			})
 		})
 
 		t.Run(fmt.Sprintf("Sqlinstances %s", testCase.name), func(t *testing.T) {

